components/compositing/scrolling.rs

```diff
@@ -55,9 +57,13 @@ impl ScrollingTimerProxy {
 impl ScrollingTimer {
     pub fn run(&mut self) {
         while let Ok(ToScrollingTimerMsg::ScrollEventProcessedMsg(timestamp)) = self.receiver.recv() {
-            let target = timestamp as i64 + TIMEOUT;
-            let delta_ns = target - (time::precise_time_ns() as i64);
-            sleep_ms((delta_ns / 1000000) as u32);
+            const NANOS_PER_SEC: u64 = 1_000_000_000;
+            let target = timestamp + TIMEOUT;
+            let delta_ns = target - (time::precise_time_ns());
+            let delta_secs = delta_ns / NANOS_PER_SEC;
+            let delta_nanos = delta_ns % NANOS_PER_SEC;
+            assert!(delta_nanos <= u32::MAX as u64);
+            thread::sleep(Duration::new(delta_secs, delta_nanos as u32));
```

Comment:
I think we could also have a function for this in `util/time` (for future usage), which takes nanoseconds as input and returns a duration.